### PR TITLE
Add: rpmsg tests for i.MX and TI processors

### PR DESCRIPTION
**Pull request by stanley31huang**
adding rpmsg tests for i.MX and TI processors

#### Test submissions
https://certification.canonical.com/hardware/202212-30918/submission/332376/

#### Unit test results
```
stanley@stanley-ThinkPad-T490:~/Desktop/Git_Repos/iot/checkbox-provider-ce-oem/checkbox-provider-ce-oem/tests$ python3 -m unittest -v
test_arm_processor_am62 (test_rpmsg_tests.RpmsgTests)
Checking if the CPU processor manufacture is TI ... SoC family is AM62X, machine is Texas Instruments AM625 SK
ok
test_arm_processor_imx (test_rpmsg_tests.RpmsgTests)
Checking if the CPU processor manufacture is i.MX ... SoC family is Freescale i.MX, machine is Honeywell i.MX8MM X8Med35 PDK
ok
test_arm_processor_unexpected (test_rpmsg_tests.RpmsgTests)
/sys/devices/soc0/family and machine fils not exists ... SoC family is , machine is 
ok
test_rpmsg_device_exist (test_rpmsg_tests.RpmsgTests)
Checking RPMSG device is available ... ## Checking RPMSG device is available ...
RPMSG device is available
ok
test_rpmsg_device_not_exist (test_rpmsg_tests.RpmsgTests)
Checking RPMSG device is not available ... ## Checking RPMSG device is available ...
ok

----------------------------------------------------------------------
Ran 5 tests in 0.004s

OK
```

### CODE REVIEW

**Review by LiaoU3 — CHANGES_REQUESTED** [review]
@stanley31huang Please see my in-line comment.

File: checkbox-provider-ce-oem/bin/rpmsg_tests.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    Read data from /sys/devices/soc0/machine
+
+    Returns:
+        soc_machine (str): SoC machine.
+    """
+    soc_machine = ""
+    path = "/sys/devices/soc0/machine"
+    if os.path.isfile(path):
+        with open(path, "r") as fp:
+            soc_machine = fp.read().strip()
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/rpmsg_tests.py:92]: the scripts will raise a SystemExit in pingpong function. And we would like to collect both family and machine while doing the testing
```
SystemExit("Unexpected CPU type.")
```

File: checkbox-provider-ce-oem/bin/rpmsg_tests.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    Read data from /sys/devices/soc0/family
+
+    Returns:
+        soc_family (str): SoC family.
+    """
+    soc_family = ""
+    path = "/sys/devices/soc0/family"
+    if os.path.isfile(path):
+        with open(path, "r") as fp:
+            soc_family = fp.read().strip()
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/rpmsg_tests.py:75]: the scripts will raise a SystemExit in pingpong function. And we would like to collect both family and machine while doing the testing
```
SystemExit("Unexpected CPU type.")
```

File: checkbox-provider-ce-oem/bin/rpmsg_tests.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+def get_soc_family():
+    """
+    Read data from /sys/devices/soc0/family
+
+    Returns:
+        soc_family (str): SoC family.
+    """
+    soc_family = ""
+    path = "/sys/devices/soc0/family"
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/rpmsg_tests.py]: fixed

File: checkbox-provider-ce-oem/bin/rpmsg_tests.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+def get_soc_machine():
+    """
+    Read data from /sys/devices/soc0/machine
+
+    Returns:
+        soc_machine (str): SoC machine.
+    """
+    soc_machine = ""
+    path = "/sys/devices/soc0/machine"
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/rpmsg_tests.py]: fixed

File: checkbox-provider-ce-oem/bin/rpmsg_tests.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        arm_cpu_type (str): ARM CPU type. E.g. ti, imx
+    """
+    family = get_soc_family()
+    machine = get_soc_machine()
+    print("SoC family is {}, machine is {}".format(family, machine))
+
+    if family.find("i.MX") != -1 or machine.find("i.MX") != -1:
+        arm_cpu_type = "imx"
+    else:
+        if machine.find("Texas Instruments") != -1:
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/rpmsg_tests.py]: fixed

File: checkbox-provider-ce-oem/bin/rpmsg_tests.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (LiaoU3) [reply checkbox-provider-ce-oem/bin/rpmsg_tests.py:93]: Copy that

File: checkbox-provider-ce-oem/bin/rpmsg_tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (LiaoU3) [reply checkbox-provider-ce-oem/bin/rpmsg_tests.py:76]: Copy that

**Review by LiaoU3 — APPROVED** [review]
LGTM

File: checkbox-provider-ce-oem/units/rpmsg/jobs.pxu [hunk 6]
```diff
@@ -0,0 +1,22 @@
+id: ce-oem-rpmsg/detect-device
+_summary: Check if the RPMSG devices is initialed.
+category_id: rpmsg
+estimated_duration: 60
```
Comment (patliuu) [inline checkbox-provider-ce-oem/units/rpmsg/jobs.pxu:4]: Detect job should not need 60 seconds, right? :smile:

**Review by patliuu — CHANGES_REQUESTED** [review]

**Review by patliuu — APPROVED** [review]
LGTM

**Pull request merged**